types/shared.d.ts

```diff
@@ -0,0 +1,135 @@
+export type GenericCallback = (...arg: any) => void;
+
+export interface ViSearchClient {
+  q: any;
+  set: (key: string, value: unknown) => void;
+  set_keys: (keys: Record<string, unknown>) => void;
+  send: (
+    action: string,
+    eventParams: Record<string, unknown>,
+    callback?: (action: string, params: Record<string, unknown>) => void,
+    failure?: GenericCallback
+  ) => void;
+  send_events: (
+    action: string,
+    eventParamsList: Record<string, unknown>[],
+    callback?: () => void,
+    failure?: GenericCallback
+  ) => void;
+  product_search_by_image: (
+    params: Record<string, unknown>,
+    callback: (resp: PSResponse) => void,
+    failure?: GenericCallback
+  ) => Promise<void>;
+  product_search_by_id: (
+    pid: string,
+    params: Record<string, unknown>,
+    callback: (resp: PSResponse) => void,
+    failure?: GenericCallback
+  ) => Promise<void>;
+  product_recommendations: (
+    pid: string,
+    params: Record<string, unknown>,
+    callback: (resp: PSResponse) => void,
+    failure?: GenericCallback
+  ) => Promise<void>;
+  product_recommendations_by_post: (
+    pid: string,
+    params: Record<string, unknown>,
+    callback: (resp: PSResponse) => void,
+    failure?: GenericCallback
+  ) => Promise<void>;
+  product_search_by_id_by_post: (
+    pid: string,
+    params: Record<string, unknown>,
+    callback: (resp: PSResponse) => void,
+    failure?: GenericCallback
+  ) => Promise<void>;
+  set_uid: (uid: string, callback?: (uid: string) => void, failure?: GenericCallback) => void;
+  get_uid: (callback: (uid: string) => void, failure?: GenericCallback) => void;
+  get_sid: (callback: (sid: string) => void, failure?: GenericCallback) => void;
+  get_last_query_id: (callback: (lastQueryId: string | null) => void, failure?: GenericCallback) => void;
+  get_session_time_remaining: (callback: (time: number) => void, failure?: GenericCallback) => void;
+  reset_session: (callback?: (sid: string) => void, failure?: GenericCallback) => void;
+  get_default_tracking_params: (callback: (params: Record<string, unknown>) => void, failure?: GenericCallback) => void;
+  apply_prototypes_call: (command: [keyof ViSearchClient, unknown]) => void;
+  resize_image: (
+    imageAsDataUrl: string,
+    resizeSettings: PSResizeSettings | undefined,
+    onSuccess: GenericCallback,
+    onFailure?: GenericCallback
+  ) => Promise<void>;
+  generate_uuid: (callback: (uuid: string) => void, failure?: GenericCallback) => void;
```

Comment:
I believe we should convert these method names to `camelCase` now. If we bump the version number, we have the liberty to introduce breaking changes.
Another compelling reason: once converted, the methods `send` (which should be renamed `sendEvent`), `sendEvents`, the UID and session-related methods should be reminiscent of another interface already defined somewhere.